browser/flagr-ui/src/components/Flag.vue

```diff
@@ -198,6 +198,23 @@
                     </el-col>
                   </el-row>
                 </el-card>
+                <el-card>
+                  <div slot="header" class="el-card-header">
```

Comment:
This card header slot has a strong background color, maybe you can just move the inner divs outside.

![image](https://user-images.githubusercontent.com/658840/54876220-751b4f00-4dc9-11e9-8c59-aeb258444249.png)

Reply:
Something like this?

![Screen Shot 2019-03-24 at 12 55 16 PM](https://user-images.githubusercontent.com/5851331/54882797-18a14980-4e34-11e9-98ce-c17ffd06d064.png)

Reply:
```
-                </el-card>
-                <el-card>
-                  <div slot="header" class="el-card-header">
-                    <div class="flex-row">
-                      <div class="flex-row-left">
-                        <h2>Flag Notes</h2>
-                      </div>
-                      <div class="flex-row-right">
-                        <el-button circle style="float: right;" @click="toggleShowMdEditor" :class="editViewIcon"/>
-                      </div>
-                    </div>
-                  </div>
-                  <markdown-editor
-                    :showEditor="this.showMdEditor"
-                    :markdown.sync="flag.notes"
-                    @save="putFlag(flag)"
-                  ></markdown-editor>



+                  <el-row>
+                    <h5>
+                      <span>Flag Note</span>
+                      <el-button round size="mini" @click="toggleShowMdEditor">
+                        <span class="el-icon-edit"></span> edit
+                      </el-button>
+                    </h5>
+                  </el-row>
+                  <el-row>
+                    <markdown-editor
+                      :showEditor="this.showMdEditor"
+                      :markdown.sync="flag.notes"
+                      @save="putFlag(flag)"
+                      ></markdown-editor>
+                  </el-row>
```

Something like, I think we need to move the markdown editor inside the previous card, since "save flag" button need to cover the note part. Let me know what you think.

![image](https://user-images.githubusercontent.com/658840/54964827-271e5c80-4f2b-11e9-9ed3-63abe1507f79.png)

Reply:
Ah, I see what you mean. That looks pretty good to me, I am happy to make the change. It makes sense as notes are logically part of the flag entity.

I will note that while the "save flag" button does indeed save notes (as it should) there is also a save button within the editor itself which will save the flag. At first I was thinking whether I should disable the save button within the editor as it is a little weird with the redundancy but I decided to keep it for two reasons:
1) Reusability (if we want to reuse the component we may not have a "save flag"-esque button). It is true that we could make the save button optional and disabled in this case, but that leads me to the next reason
2) One of the editing modes is full-screen which is great for longer notes. I wanted the option to be able to save from that view.

That being said, I still think it makes sense moving the editor into the flag card given that they are part of the note, but let me know what you think.

Also, as a side note, I experimented a bit with adding an autosave feature so that the flag would be saved either every x seconds while editing or after a certain amount of changes were made since last save (save flag notice disabled in this mode). While it worked great, it made the snapshots get quite messy and I didn't have a great solution for that. I decided to drop that feature as I didn't (and still don't) think it is worth the complication but if you have any good ideas related to handling the snapshots that I'd be curious to hear them.

Reply:
Thanks! I like the save button and the full-screen mode. Yes, autosave doesn't play well with snapshots and diffs.

Reply:
If you really want autosave, we can exclude note from the snapshot.